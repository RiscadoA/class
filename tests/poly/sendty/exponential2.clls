proc main() {
    cut {
        println("!"); !x(y);
        println("sendty"); sendty y(lint);
        println("lhs send"); send y(a. println("let"); let a 2);
        println(y); ()
        |x: ? recvty T; recv ~T; T |
        println("?"); ?x;
        println("call"); call x(y);
        println("recvty"); recvty y(T);
        println("rhs recv"); recv y(a);
        println("fwd"); fwd a y
    }
};;
