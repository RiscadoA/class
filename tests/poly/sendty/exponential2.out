!
?
call
sendty
recvty
lhs send
let
rhs recv
fwd
2
